server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndicesStatsResponse.java

```diff
@@ -83,26 +82,18 @@ public IndexStats getIndex(String index) {
         if (indicesStats != null) {
             return indicesStats;
         }
-        Map<String, IndexStats> indicesStats = new HashMap<>();
 
-        Set<Index> indices = new HashSet<>();
+        final Map<String, IndexStatsBuilder> indicesStatsBuilder = new HashMap<>();
         for (ShardStats shard : shards) {
-            indices.add(shard.getShardRouting().index());
-        }
-
-        for (Index index : indices) {
-            List<ShardStats> shards = new ArrayList<>();
+            Index index = shard.getShardRouting().index();
             String indexName = index.getName();
```

Comment:
since this is only used once now, does it make sense to call `index.getName()` inline and remove this local variable?